### PR TITLE
feat(apig): add new datasource to get list of channels

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**:
feat(apig): add new datasource to get list of channels.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/apig" TESTARGS="-run TestAccDataSourceChannels_basic"
...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccDataSourceChannels_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceChannels_basic
=== PAUSE TestAccDataSourceChannels_basic
=== CONT  TestAccDataSourceChannels_basic
--- PASS: TestAccDataSourceChannels_basic (616.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      616.844s
```

### CODE REVIEW

File: docs/data-sources/apig_channels.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `instance_id` - (Required, String) Specifies the ID of the dedicated instance to which the channels belong.
+
+* `channel_id` - (Optional, String) Specifies the ID of the to be queried.
```
Comment (ruwenqiang123) [inline docs/data-sources/apig_channels.md]:
```suggestion
* `channel_id` - (Optional, String) Specifies the ID of the VPC channel to be queried.
```

File: docs/data-sources/apig_channels.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+* `id` - The data source ID.
+
+* `vpc_channels` - All VPC channels that match the filter parameters.
+  The [vpc_channels](#vpc_channels) structure is documented below.
+
+<a name="vpc_channels"></a>
+The `vpc_channels` block supports:
+
+* `id` - The ID of the vpc channel.
```
Comment (ruwenqiang123) [inline docs/data-sources/apig_channels.md]:
```suggestion
* `id` - The ID of the VPC channel.
```

Comment (ruwenqiang123) [reply docs/data-sources/apig_channels.md]: please check other parts of the document.

File: docs/data-sources/apig_channels.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+* `balance_strategy` - The distribution algorithm.
+
+* `member_type` - The member type of the VPC channel.
+
+* `type` - The type of the VPC channel.
+
+* `created_at` - The creation time of channel, in RFC3339 format.
+
+* `member_group` - The parameter member groups of the VPC channels..
```
Comment (ruwenqiang123) [inline docs/data-sources/apig_channels.md]:
```suggestion
* `member_group` - The parameter member groups of the VPC channels.
```

File: docs/data-sources/apig_channels.md (hunk 1 above)
Comment (wangyuancheng01) [reply docs/data-sources/apig_channels.md]: done

File: docs/data-sources/apig_channels.md (hunk 2 above)
Comment (wangyuancheng01) [reply docs/data-sources/apig_channels.md]: done

File: docs/data-sources/apig_channels.md (hunk 3 above)
Comment (wangyuancheng01) [reply docs/data-sources/apig_channels.md]: done

File: huaweicloud/provider.go [hunk 4]
```diff
@@ -407,6 +407,7 @@ func Provider() *schema.Provider {
 			"huaweicloud_apig_api_associated_throttling_policies": apig.DataSourceApiAssociatedThrottlingPolicies(),
 			"huaweicloud_apig_api_basic_configurations":           apig.DataSourceApiBasicConfigurations(),
 			"huaweicloud_apig_custom_authorizers":                 apig.DataSourceCustomAuthorizers(),
+			"huaweicloud_apig_channels":                           apig.DataSourceChannels(),
```
Comment (Lance52259) [inline huaweicloud/provider.go]:
```suggestion
			"huaweicloud_apig_channels":                           apig.DataSourceChannels(),
			"huaweicloud_apig_custom_authorizers":                 apig.DataSourceCustomAuthorizers(),
```

File: huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_channels_test.go [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+			{
+				Config: testAccDataSourceChannels_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestMatchResourceAttr(dataSource, "vpc_channels.#", regexp.MustCompile(`^[1-9]([0-9]*)?$`)),
+					resource.TestCheckResourceAttrSet(dataSource, "vpc_channels.0.id"),
+					resource.TestCheckResourceAttrSet(dataSource, "vpc_channels.0.name"),
+					resource.TestCheckResourceAttrSet(dataSource, "vpc_channels.0.member_group.0.id"),
+					resource.TestCheckResourceAttrSet(dataSource, "vpc_channels.0.member_group.0.name"),
+					dcById.CheckResourceExists(),
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_channels_test.go]: pls check the time format

File: huaweicloud/provider.go (hunk 4 above)
Comment (wangyuancheng01) [reply huaweicloud/provider.go]: done

File: huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_channels_test.go (hunk 5 above)
Comment (wangyuancheng01) [reply huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_channels_test.go]: done

**Pull request merged**